### PR TITLE
Basic setup of final scene and scripts for it.

### LINKED ISSUE #807: Scripting final scene
Create the scripts for the final scene.

### PR DESCRIPTION
**Pull request by andens**
Scripts doesn't do anything particular during the losing condition because it's not specified. After killing the monster we should fade out but other than that everything is in place.

Resolves #807

### CODE REVIEW

File: Game/Resources/final_event/MonsterScript.as [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+				if (waitTimer >= 1.0f) {
+					phase = 3;
+				}
+				break;
+			}
+			case 3: { // Approach player
+				self.position.x -= 6.0f * deltaTime;
+				break;
+			}
+			case 5: { // Eating
```
Comment (LarsIngo) [inline Game/Resources/final_event/MonsterScript.as]: case 4?

File: Game/Resources/final_event/MonsterScript.as [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	}
+
+	void BecomeExposed() {
+		print("I am becoming exposed now!\n");
+		phase = 1;
+	}
+
+	void StopCharging() {
+		print("Stopping my charge\n");
+		phase = 4;
```
Comment (LarsIngo) [inline Game/Resources/final_event/MonsterScript.as]: case 4?

File: Game/Resources/final_event/MainScript.as [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	Entity @monster;
+    Entity @rightHand;
+	float waitForMonsterTimer = 0.0f;
+    bool knifePickedUp = false;
+
+    MainScript(Entity @entity){
+        @hub = Managers();
+        @self = @entity;
+
+		@minecart = GetEntityByGUID(1508919163);
```
Comment (LarsIngo) [inline Game/Resources/final_event/MainScript.as]: Remove tabs

File: Game/Resources/final_event/MonsterScript.as (hunk 1 above)
Comment (andens) [reply Game/Resources/final_event/MonsterScript.as]: The fourth phase is when the monster waits for the player to enter its warm embrace (transitioned into via `StopCharging`). It has no special logic which is why there is no case for it.

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 8 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L115](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c92aa82bb4160780c6269038962a5ee24c59a948/src/Engine/Manager/ScriptManager.cpp#L115): Variable 'callstack' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L116](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c92aa82bb4160780c6269038962a5ee24c59a948/src/Engine/Manager/ScriptManager.cpp#L116): Variable 'variables' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L609](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c92aa82bb4160780c6269038962a5ee24c59a948/src/Engine/Manager/ScriptManager.cpp#L609): The function 'ClearBreakpoints' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L821](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c92aa82bb4160780c6269038962a5ee24c59a948/src/Engine/Manager/ScriptManager.cpp#L821): The function 'GetStringDeclarationID' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L827](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c92aa82bb4160780c6269038962a5ee24c59a948/src/Engine/Manager/ScriptManager.cpp#L827): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [TriggerManager.cpp#L100](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c92aa82bb4160780c6269038962a5ee24c59a948/src/Engine/Manager/TriggerManager.cpp#L100): The function 'GetTriggerComponents' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [TriggerRepeat.cpp#L123](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c92aa82bb4160780c6269038962a5ee24c59a948/src/Engine/Trigger/TriggerRepeat.cpp#L123): The function 'GetOwningEntity' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 6 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L115](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c6809a47b789c3bbebb1bd90e8810391be176ace/src/Engine/Manager/ScriptManager.cpp#L115): Variable 'callstack' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L116](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c6809a47b789c3bbebb1bd90e8810391be176ace/src/Engine/Manager/ScriptManager.cpp#L116): Variable 'variables' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L609](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c6809a47b789c3bbebb1bd90e8810391be176ace/src/Engine/Manager/ScriptManager.cpp#L609): The function 'ClearBreakpoints' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L821](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c6809a47b789c3bbebb1bd90e8810391be176ace/src/Engine/Manager/ScriptManager.cpp#L821): The function 'GetStringDeclarationID' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L827](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c6809a47b789c3bbebb1bd90e8810391be176ace/src/Engine/Manager/ScriptManager.cpp#L827): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**